rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/model/RequestPayload.scala

```diff
@@ -0,0 +1,36 @@
+package org.elasticmq.rest.sqs.model
+
+import akka.http.scaladsl.server.Directive0
+import akka.http.scaladsl.server.Directives._
+import org.elasticmq.rest.sqs.{Action, FlatParamsReader}
+import org.elasticmq.rest.sqs.model.RequestPayload.QueryParams
+import spray.json.{JsObject, RootJsonFormat}
+
+sealed trait RequestPayload {
+  def action(requiredAction: Action.Value): Directive0
+
+  def maybeAction: Option[String]
```

Comment:
The method name doesn't sound right to me. It doesn't really say what's the meaning of that method even if the implementation is passed to class implementing that Trait.

It's used in two places:
```
override def maybeAction: Option[String] = params.get("Action") //Map[String, String]
override def maybeAction: Option[String] = Some(action)
```

In first we're `getting` an `Action` and in second we're wrapping it but the end result is us retrieving an optinal value of Action, so maybe `retrieveAction` might be a better name here?